packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5433,19 +5421,24 @@ class InheritedElement extends ProxyElement {
 /// ### Specializing the getters
 ///
 /// [RenderObjectElement] objects spend much of their time acting as
-/// intermediaries between their [widget] and their [renderObject]. To make this
-/// more tractable, most [RenderObjectElement] subclasses override these getters
-/// so that they return the specific type that the element expects, e.g.:
+/// intermediaries between their [widget] and their [renderObject]. It is
+/// generally recommended against specializing the [widget] getter and
+/// instead casting at the various callsites to avoid adding overhead
+/// outside of this particular implementation. If the [widget] getter
+/// is widely used, a separte getter that specializes the Widget type
+/// should be used instead.
 ///
 /// ```dart
 /// class FooElement extends RenderObjectElement {
 ///
 ///   @override
-///   Foo get widget => super.widget as Foo;
-///
-///   @override
 ///   RenderFoo get renderObject => super.renderObject as RenderFoo;
 ///
+///   void _foo() {
+///     final Foo foo = widget as Foo;
+///   }
+///
+///   Foo get fooWidget => widget as Foo;
```

Comment:
Maybe leave a comment above this line to make clear that this is an alternative approach to casting at the callsite to avoid the impression that its best practice to have this getter.

Reply:
Or maybe remove this line altogether. I think best practice should be to just cast at the callsite.

Reply:
done